File: src/main/java/com/dsa/problems/scaler/hashing/check_pair_sum.java
package com.dsa.problems.scaler.hashing;

import java.util.*;

public class check_pair_sum {
  /**
   * Q. Check Pair Sum
   *
   * Given an Array of integers A, and a target sum B.
   * Check if there exists a pair (i,j) such that Ai + Aj = B and i!=j.
   */
  private static boolean pairSum(ArrayList<Integer> A, int B) {
    HashSet<Integer> hs = new HashSet<>();
    for(int i = 0; i < A.size(); i++) {
      final int comp = B - A.get(i);
      if(hs.contains(comp)) {
        return true;
      }
      hs.add(A.get(i));
    }

    return false;
  }

  public static void main(String[] args) {
    System.out.println(pairSum(new ArrayList<>(Arrays.asList(-1, 2, 3, 2, 1)), 3));
  }
}

File: problems/scaler/hashing/check_pair_sum.js
/**
 * Q. Check Pair Sum
 *
 * Given an Array of integers B, and a target sum A.
 * Check if there exists a pair (i,j) such that Bi + Bj = A and i!=j.
 */
function soln(A, B) {
  const hm = new Set();

  for (let i = 0; i < A.length; i++) {
    const comp = B - A[i];
    if (hm.has(comp)) {
      return 1;
    } else {
      hm.add(A[i]);
    }
  }

  return 0;
}

console.log(soln([1, 2, 3, 4, 5], 9));
console.log(soln([23, 52, 95, 29, 9, 47, 49, 55, 96, 4], 100));
